src/mds/Server.cc

```diff
@@ -10333,7 +10333,7 @@ void Server::handle_client_lssnap(MDRequestRef& mdr)
 
     encode(snap_name, dnbl);
     //infinite lease
-    LeaseStat e(CEPH_LEASE_VALID, -1, 0);
+    LeaseStat e(CEPH_LEASE_VALID, -1, 0, p->second->ino != diri->ino());
```

Comment:
does LeastStat's constructor accept four arguments?

Reply:
Sorry, forgot to amend that change, I will fix it soon.

Reply:
Done.